storage3/_async/file_api.py

```diff
@@ -201,16 +201,37 @@ async def create_signed_urls(
             ] = f"{self._client.base_url}{cast(str, item['signedURL']).lstrip('/')}"
         return data
 
-    async def get_public_url(self, path: str, options: TransformOptions = {}) -> str:
+    async def get_public_url(self, path: str, options: URLOptions = {}) -> str:
         """
         Parameters
         ----------
         path
             file path, including the path and file name. For example `folder/image.png`.
         """
+        _query_string = []
+        download_query = None
+        if options.get("download"):
+            download_query = (
+                "download="
+                if options.get("download") is True
+                else f"download={options.get('download')}"
+            )
+
+        if download_query:
+            _query_string.append(download_query)
+
         render_path = "render/image" if options.get("transform") else "object"
-        transformation_query = urllib.parse.urlencode(options)
-        query_string = f"?{transformation_query}" if transformation_query else ""
+        transformation_query = (
+            urllib.parse.urlencode(options.get("transform"))
+            if options.get("transform")
+            else None
+        )
+
+        if transformation_query:
+            _query_string.append(transformation_query)
+
+        query_string = "&".join(_query_string)
+        query_string = f"?{query_string}"
```

Comment:
A little confused by this diff - why are we manually constructing the query string instead of letting urllib.parse.urlencode do the work?